java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -577,17 +580,23 @@ public void visitNewClass(NewClassTree tree) {
       }
       scan(classBody);
       registerType(tree, anonymousClassType);
+      // } else if (callToInitWithDiamondOperator(tree.identifier()) && !resolution.type().isUnknown()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)